core/src/main/java/org/neo4j/graphalgo/core/sources/SingleRunAllRelationIterator.java

```diff
@@ -0,0 +1,52 @@
+package org.neo4j.graphalgo.core.sources;
+
+import org.neo4j.graphalgo.api.*;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.api.ReadOperations;
+import org.neo4j.kernel.impl.core.ThreadToStatementContextBridge;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+
+import java.util.Iterator;
+
+/**
+ *
+ * @author mknblch
+ */
+public class SingleRunAllRelationIterator implements AllRelationIterator, AutoCloseable {
+
+    private final GraphDatabaseAPI api;
+    private final Transaction transaction;
+    private final IdMapping idMapping;
+
+    public SingleRunAllRelationIterator(GraphDatabaseAPI api, IdMapping idMapping) {
+        this.api = api;
+        transaction = api.beginTx();
```

Comment:
has to be ensured that this is only used in a single thread, better handle the transaction in the forEachRel... method